src/appleseed/renderer/modeling/environmentedf/oslenvironmentedf.cpp

```diff
@@ -76,11 +151,17 @@ namespace
     {
       public:
         OSLEnvironmentEDF(
-            const char*             name,
-            const ParamArray&       params)
+            const char*         name,
+            const ParamArray&   params)
           : EnvironmentEDF(name, params)
+          , m_importance_map_size(0)
+          , m_probability_scale(0.0f)
         {
             m_inputs.declare("osl_background", InputFormatEntity, "");
+
+            // TODO:
+            // Add optional values from m_inputs (see LatLongMapEnvironmentEDF).
+            m_importance_map_size = m_params.get_optional<size_t>("importance_map_size", 1024);
```

Comment:
I think this would be best done in `on_render_begin()`.

Reply:
Agree. Done.